include/deal.II/meshworker/assemble_flags.h

```diff
@@ -0,0 +1,203 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__mesh_worker_assemble_flags_h
+#define dealii__mesh_worker_assemble_flags_h
+
+
+#include <deal.II/base/config.h>
+#include <vector>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/*!@addtogroup MeshWorker */
+/*@{*/
+
+namespace MeshWorker
+{
+
+  /**
+   * The enum type given to the mesh_loop() function, telling that function which
+   * elements need to be assembled.
+   *
+   * You can select more than one flag by concatenation using the bitwise or
+   * `operator|(AssembleFlags,AssembleFlags)`.
+   *
+   * @author Luca Heltai, 2017.
+   */
+  enum AssembleFlags
+  {
+    /**
+     * Do Nothing.
+     */
+    assemble_nothing = 0,
+    /**
+     * Assemble on locally owned cells.
+     */
+    assemble_own_cells = 0x0001,
+    /**
+     * Assemble on ghost cells.
+     */
+    assemble_ghost_cells = 0x0002,
+    /**
+     * Assemble on interior faces between two locally owned cells,
+     * visiting each face only once
+     */
+    assemble_own_interior_faces_once = 0x0004,
+    /**
+     * Assemble on interior faces between two locally owned cells,
+     * visiting each interior face twice
+     */
+    assemble_own_interior_faces_both = 0x0008,
+    /**
+     * Assemble on faces between a locally owned cell and a ghost cell, making
+     * sure that only one of the processes will assemble these faces (from the
+     * finer side or the process with the lower mpi rank)
+     */
+    assemble_ghost_faces_once = 0x0010,
+    /**
+     * Assemble on faces between a locally owned cell and a ghost cell, both
+     * processes will assemble these faces. Note that these faces are never
+     * assembled from both sides on a single process.
+     */
+    assemble_ghost_faces_both = 0x0020,
+    /**
+     * Assemble on boundary faces of the locally owned cells
+     */
+    assemble_boundary_faces = 0x0040,
+
+    /**
+     * Assemble cell integrals before face integrals.
+     */
+    cells_first = 0x0080,
```

Comment:
if you don't set this flag, does the *opposite* happen? If so, maybe document that as well.